instrumentation/github.com/confluentinc/confluent-kafka-go/kafka/splunkkafka/test/kafka_test.go

```diff
@@ -47,7 +46,54 @@ var (
 	testTopic   = "gotest"
 )
 
-func TestSynchronous(t *testing.T) {
+func TestChannelBasedProducer(t *testing.T) {
+	defer goleak.VerifyNone(t)
+
+	partition := int32(0)
+	sr, opts := newFixtures()
+	p := newProducer(t, opts...)
+
+	done := make(chan struct{})
+	var sent *kafka.Message
+	go func() {
+		defer close(done)
+		sent = requireEventIsMessage(t, <-p.Events())
```

Comment:
we should not use `require` or `t.Fatalf/FailNow` in new goroutines.
See: https://pkg.go.dev/testing#T.FailNow

> FailNow must be called from the goroutine running the test or benchmark function, not from other goroutines created during the test. Calling FailNow does not stop those other goroutines.

I prefer having all assertions in the main testing goroutine if it does not complicate the code too much. But usually it s nicer to have all assertions grouped.

When looking at this code I am not even sure if spawning a new goroutine to receive the message is needed at all